pkg/apis/batch/types.go

```diff
@@ -44,13 +44,23 @@ const (
 	JobNameLabel = labelPrefix + LegacyJobNameLabel
 	// Controller UID is used for selectors and labels for jobs
 	ControllerUidLabel = labelPrefix + LegacyControllerUidLabel
+	// Label indicating the controller that manages a Job. Jobs with a custom value
+	// of this label - any label value other than `job-controller.k8s.io` - are
+	// skipped by the job controller, and their reconciliation is delegated to
+	// an external controller, indicated by the value of the label. Jobs that
+	// don't have this label at all or where the label value is the reserved
+	// string `job-controller.k8s.io`, are reconciled by the built-in job controller.
+	JobManagedByLabel = labelPrefix + "managed-by"
 	// Annotation indicating the number of failures for the index corresponding
 	// to the pod, which are counted towards the backoff limit.
 	JobIndexFailureCountAnnotation = labelPrefix + "job-index-failure-count"
 	// Annotation indicating the number of failures for the index corresponding
 	// to the pod, which don't count towards the backoff limit, according to the
 	// pod failure policy. When the annotation is absent zero is implied.
 	JobIndexIgnoredFailureCountAnnotation = labelPrefix + "job-index-ignored-failure-count"
+	// JobControllerName reserved value for the managed-by label for the built-in
```

Comment:
In the doc updates, let's make sure to document this label.

Reply:
https://kubernetes.io/docs/reference/labels-annotations-taints/

Reply:
Makes sense, I added it already to the WIP docs PR: https://github.com/kubernetes/website/pull/45132/files#diff-3e60cf1d53572377993fc624af259e045db43b85ec5343e146ca46d1fd56a507R1203, so that we don't forget